test/common/grpc/async_client_manager_impl_test.cc

```diff
@@ -207,21 +245,57 @@ TEST_F(AsyncClientManagerImplTest, GrpcServiceConfigWithHashKeyTest) {
             config_with_hash_key_c.getPreComputedHash());
 }
 
+TEST_F(AsyncClientManagerImplTest, RawAsyncClientCacheWithConfig) {
+  envoy::config::core::v3::GrpcService grpc_service;
+  grpc_service.mutable_envoy_grpc()->set_cluster_name("foo");
+
+  Bootstrap::GrpcAsyncClientManagerConfig aync_manager_config;
+  aync_manager_config.mutable_entry_expiration_time()->MergeFrom(
+      ProtobufUtil::TimeUtil::SecondsToDuration(20));
+
+  createAsyncClientManager(aync_manager_config);
```

Comment:
The refactor here causes some confusion, because a default async-client-mgr is always created, and then it is overridden here.
A better design will probably be having an `initialize(Opt<Config>)` method that needs to be called to initialize the async-client-mgr correctly.

Reply:
Yes, my initial thought is to avoid many initialize() calls for other tests. Modified.